src/main/java/com/example/bookstore/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,34 @@
+package com.example.bookstore.service.impl;
+
+import com.example.bookstore.dto.user.UserRegistrationRequestDto;
+import com.example.bookstore.dto.user.UserResponseDto;
+import com.example.bookstore.exception.RegistrationException;
+import com.example.bookstore.mapper.UserMapper;
+import com.example.bookstore.model.User;
+import com.example.bookstore.repository.user.UserRepository;
+import com.example.bookstore.service.UserService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+
+    @Override
+    public UserResponseDto register(UserRegistrationRequestDto requestDto)
+            throws RegistrationException {
+        if (userRepository.findByEmail(requestDto.getEmail()).isPresent()) {
```

Comment:
It's better to use this implementation, to avoid calling isPresent()
```suggestion
        if (userRepo.existsByEmail(request.email())) {
            throw new RegistrationException(
                    String.format("User with this email: %s already exists", request.email())
            );
        }
```